storage_controller/src/reconciler.rs

```diff
@@ -848,14 +848,19 @@ impl Reconciler {
     }
 
     /// Keep trying to notify the compute indefinitely, only dropping out if:
-    /// - the node `origin` becomes unavailable
-    /// - the node `origin` no longer has our tenant shard attached
-    /// - our cancellation token fires
+    /// - the node `origin` becomes unavailable -> Ok(())
+    /// - the node `origin` no longer has our tenant shard attached -> Ok(())
+    /// - our cancellation token fires -> Err(ReconcileError::Cancelled)
```

Comment:
>Why do we return different things for them?

The intent is that we don't.  If the node becomes unavailable while sending location_conf we should hit this branch:
```
                Some(Err(e)) => {
                    match e {
                        mgmt_api::Error::Cancelled => {
                            tracing::info!(
                                "Giving up on compute notification because {origin} is unavailable"
                            );
                            return Ok(());
                        }
```

Whereas if with_client_retries returns None, it indicates that Reconciler::cancel has fired, and we hit this branch:
```
                None => return Err(ReconcileError::Cancel),
```

Reply:
Ah. `our cancellation token` is the reconciler's cancellation token. Makes sense now!